a3p-integration/proposals/p:upgrade-19/provisionPool.test.js

```diff
@@ -0,0 +1,129 @@
+/* eslint-env node */
+/**
+ * @file The goal of this file is to make sure v28-provisionPool and v14-bank can be successfully
+ * upgraded. These vats are related because of the issues below;
+ * - https://github.com/Agoric/agoric-sdk/issues/8722
+ * - https://github.com/Agoric/agoric-sdk/issues/8724
+ *
+ * The test scenario is as follows;
+ * - Prerequisite: provisionPool and bank are already upgraded. See `upgrade.go`
+ * 1. Add a new account and successfully provision it
+ *  - Observe new account's address under `published.wallet.${address}`
+ * 2. Send some USDC_axl to provisionPoolAddress and observe its IST balances increases accordingly
+ * 3. Introduce a new asset to the chain and start a PSM instance for the new asset
+ * 3a. Deposit some of that asset to provisionPoolAddress
+ * 3b. Observe provisionPoolAddress' IST balance increase by the amount deposited in step 3a
+ */
+
+import '@endo/init';
+import test from 'ava';
+import { execFileSync } from 'node:child_process';
+import {
+  addUser,
+  makeAgd,
+  evalBundles,
+  agd as agdAmbient,
+  agoric,
+  bankSend,
+  getISTBalance,
+} from '@agoric/synthetic-chain';
+import { NonNullish } from './test-lib/errors.js';
+import {
+  retryUntilCondition,
+  waitUntilAccountFunded,
+  waitUntilContractDeployed,
+} from './test-lib/sync-tools.js';
+
+const PROVISIONING_POOL_ADDR = 'agoric1megzytg65cyrgzs6fvzxgrcqvwwl7ugpt62346';
+
+const ADD_PSM_DIR = 'addUsdLemons';
+const DEPOSIT_USD_LEMONS_DIR = 'depositUSD-LEMONS';
+
+const USDC_DENOM = NonNullish(process.env.USDC_DENOM);
+
+const agd = makeAgd({ execFileSync }).withOpts({ keyringBackend: 'test' });
+
+const ambientAuthority = {
+  query: agdAmbient.query,
+  follow: agoric.follow,
+  setTimeout,
+};
+
+const provision = (name, address) =>
+  agd.tx(['swingset', 'provision-one', name, address, 'SMART_WALLET'], {
+    chainId: 'agoriclocal',
+    from: 'validator',
+    yes: true,
+  });
+
+const introduceAndProvision = async name => {
+  const address = await addUser(name);
+  console.log('ADDR', name, address);
+
+  const provisionP = provision(name, address);
+
+  return { provisionP, address };
+};
+
+const getProvisionedAddresses = async () => {
+  const { children } = await agd.query([
+    'vstorage',
+    'children',
+    'published.wallet',
+  ]);
+  return children;
+};
+
+const checkUserProvisioned = addr =>
+  retryUntilCondition(
+    getProvisionedAddresses,
+    children => children.includes(addr),
+    'Account not provisioned',
+    { maxRetries: 5, retryIntervalMs: 1000, log: console.log, setTimeout },
+  );
+
+test(`upgrade provision pool`, async t => {
```

Comment:
Please check the incarnation #s of both vats. I think the tests would pass if the upgrade didn't take place since there's no change in behavior to detect.

Reply:
resolved in https://github.com/Agoric/agoric-sdk/commit/bc5a94ea6fe566b490104abd7773b0ad614a1c2d